client/vaultclient/vaultclient.go

```diff
@@ -256,6 +264,59 @@ func (c *vaultClient) DeriveToken(alloc *structs.Allocation, taskNames []string)
 	return tokens, nil
 }
 
+func (c *vaultClient) DeriveTokenWithJWT(ctx context.Context, reqs map[string]JWTLoginRequest) (map[string]string, error) {
```

Comment:
Because the `vault_hook` is running at the task level, it looks like `len(reqs)` is always 1 for all callers. Why have this take a map instead of the `JWTLoginRequest` alone?

Reply:
Two reasons: 

1. I thought we would need multiple Vault tokens per task. It turns out we don't 😅.
2. `DeriveToken` takes a list of tasks as well. But it also only ever gets called with a single task 🤷.

So I modified to only take a single `JWTLoginRequest` and return a `string.`